MetaMask/metamask-extension
# fix: do not show fiat amount for native asset & erc20 token if symbol is not correct

## **Description**

when you try to import a token using an incorrect symbol that refers to another token, you see an incorrect fiat amount.
this is because we don't check whether the symbol is correct or not before displaying the fiat amount.
the same bug occurs with native tokens.
the solution is to implement a new custom hooks that will check the symbol with an rpc call in the case of erc20 tokens, or using the chainid.network/chain.json api.
if the symbol is correct, we will display the fiat amount, otherwise we display nothing.

                               
-----------------------------------

**so the solution was to add custom hooks to fetch original symbol ( got from rpc calls for tokens or [chains.network](https://chainid.network/chains.json) api for native tokens) then check if it's the correct one to display the fiat amount**
-------------------------------------------------------------------------------------------------------------------------

## **Related issues**

Fixes:
 - https://github.com/MetaMask/MetaMask-planning/issues/1493

## **Manual testing steps**
- ERC20 bug
1. Import new ERC20 token ( eg: USDC )
2. change the symbol to something else ( eg: ETH )
3. you will see the balance conversion of ETH instead of USDC

- Naive token bug
1. Add new network or edit an existing one ( eg: edit Polygon network )
2. change the symbol to something else ( eg: ETH )
3. you will see the balance conversion of ETH instead of MATIC

## **Screenshots/Recordings**


### **Before**

[Native tokens](https://drive.google.com/file/d/1S7RNTNbPQd1I9Ny_wcb-O251y4wnva-n/view?usp=drive_link) 

[ERC20-tokens](https://drive.google.com/file/d/1M1XZZP8QPPiZSr5UbM3a2aJXbsDRCpfz/view?usp=drive_link)

### **After**

[Native tokens](https://drive.google.com/file/d/1Ejvr3B_usdtY6zLz8gQRKsw5yd--t80w/view?usp=drive_link)

[ERC20-tokens](https://drive.google.com/file/d/1andfAoVOU3Ovt9ULF7hK1BZbN7ROwSbP/view?usp=drive_link)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
